core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -151,8 +158,17 @@ private[server] class MetadataCache(brokerId: Int) extends Logging {
     }
   }
 
+  def getControllerId: Int = {
+    controllerId.getOrElse(MetadataResponse.NO_CONTROLLER_ID)
+  }
```

Comment:
Yeah I should return the option here and use `MetadataResponse.NO_CONTROLLER_ID` KafkaApis.